content/SCALE/SCALETutorials/Credentials/DirectoryServices/ConfigFreeIPA.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Configuring FreeIPA"
+description: "Provides information and configuration instructions for adding FreeIPA directory service in SCALE."
+weight: 25
+aliases:
+tags:
+ - ldap
+ - directoryservices
+ - freeipa
+keywords:
+ - enterprise data storage
+ - nas data storage 
+---
+
+TrueNAS provides the option to configure a FreeIPA directory service client in SCALE using the **LDAP** configuration screen under **Credentials > Directory Services**.
+
+{{< expand "Does LDAP work with SMB?" "v" >}}
+LDAP authentication for SMB shares is disabled unless you configured and populated the LDAP directory with Samba attributes.
+The most popular script for performing this task is `smbldap-tools`.
+TrueNAS needs to be able to validate the full certificate chain (no self-signed certificates).
+TrueNAS does not support non-CA certificates.
+{{< /expand >}}
+
+{{< hint type=note >}}
+You can have either Active Directory or LDAP configured on SCALE but not both.
+{{< /hint >}}
+
+## First Steps
+Your IPA server must have DNS properly configured and have access through the firewall. Having functional DNS for the IPA domain is an absolute requirement.
+If improperly configured, the IPA server cannot resolve Kerberos SRV records and the configuration in TrueNAS fails with network errors.
+Running the IPA server as the LDAP client does the realm join automatically with the ldap creds.
+
+Change the global network settings default **Hostname** setting from **truenas** to any other name. Leaving it set to the default breaks the configuration.
+
+Change the **Nameserver 1** network setting to the IP address for the IPA server and remove any other IP addresses configured as **Nameserver 2** and **Nameserver 3**.
+Having extra non-IPA DNS nameserver addresses configured in TrueNAS breaks DNS.
+
+You must set the **LDAP** screen **Hostname** to the Fully Qualified Domain Name (FQDN) for for the IPA server.
+Setting it to the IPA server IP address breaks the configuration.
+
+Check the **LDAP** screen **Base DN** setting for extraneous spaces between dn records and remove them.
+
+The LDAP service goes into a degraded state for systems with misconfigured FreeIPA.
+
+To use Kerberos for NFS/SMB authentication, do not point a random windows client at the server or SMB does not work.
+
+## Configuring LDAP
+
+To configure TrueNAS for a FreeIPA server:
+
+1. Go to **Network** and click **Configure** on the **Global Network Settings** widget to open the network settings screen.
```

Comment:
Why do we have steps under "first steps" that then appear again in the actual proceedure. To me that says "do these first steps now, before you start the procedure" but can't we just do them in order withot pulling them out to first steps?

Reply:
They are not steps, they are things  they must do. the steps are in the procedure section.